admin/sales_analytics/utils.py

```diff
@@ -0,0 +1,137 @@
+    multi_product_count = 0
+    cross_product_count = 0
+    multi_action_count = 0
+    for user_node in user_nodes:
+        if user_node['_id']:
+            user_id = user_node['_id']
+            node_id = user_node['node_id']
+            products = []
+            nodes = db.node.find({'_id': {'$in': node_id}})
+            for node in nodes:
+                products.append(get_entry_point(node['system_tags']))
+            if len(set(products)) > 1:
+                multi_product_count += 1
+
+            # Cross product count
+            user = db.user.find_one({'_id': user_id})
+            user_entry_point = get_entry_point(user['system_tags'])
+            for product in products:
+                if user_entry_point != product:
+                    cross_product_count += 1
+                    break
+
+            # Action type
+            if len(set(user_node['action'])) > 1:
+                multi_action_count += 1
+
+    return {'multi_product_count': multi_product_count,
+            'cross_product_count': cross_product_count,
+            'multi_action_count': multi_action_count,
+            }
+
+
+def get_repeat_action_user_count(db=db, timedelta=timedelta(days=30)):
+    """
+    Get the number of users that have repetitive actions (with a 3 second difference)
+    during the last month.
+    """
+    start_date = datetime.now() - timedelta
+    pipeline = [
+        {'$match': {'date': {'$gt': start_date}}},
+        {'$group': {'_id': '$user', 'nodelog_id': {'$addToSet': '$_id'}}},
+    ]
+
+    user_nodelog = db.nodelog.aggregate(pipeline)['result']
+    repeat_action_count = 0
+    repeat_action_user_age = []
+    for i in user_nodelog:
+        if i['_id']:
+            user_id = i['_id']
+            nodelog_id = i['nodelog_id']
+            nodelogs = db.nodelog.find({'_id': {'$in': nodelog_id}}).sort([('date', 1)])
+            repeat_action_date = {}
+            for nodelog in nodelogs:
+                action = nodelog['action']
+                date = nodelog['date']
+                if action not in repeat_action_date:
+                    repeat_action_date[action] = date
+                elif abs((date - repeat_action_date[action]).total_seconds()) < 3:
+                    repeat_action_date[action] = date
+                else:
+                    repeat_action_count += 1
+                    date_registered = db.user.find({'_id': user_id}).next()['date_registered']
+                    age = (date - date_registered).days
+                    repeat_action_user_age.append(age)
+                    break
+    return {'repeat_action_count': repeat_action_count, 'repeat_action_age': repeat_action_user_age}
+
+
+user_count = get_user_count(db, ENTRY_POINTS)
+multi_product_metrics_yearly = get_multi_product_metrics()
+multi_product_metrics_monthly = get_multi_product_metrics(timedelta=timedelta(days=30))
+repeat_action_user_monthly = get_repeat_action_user_count()
```

Comment:
Not convinced that this is the best way to "export" these. Basically they are run every time this file is imported, although that is only once it does also mean that you could have stale data.